master/internal/project/authz_basic_impl.go

```diff
@@ -96,6 +96,17 @@ func (a *ProjectAuthZBasic) CanMoveProject(
 	return nil
 }
 
+// CanMoveProjectExperiments returns an error if the user isn't a admin or owner of a project.
+func (a *ProjectAuthZBasic) CanMoveProjectExperiments(
+	curUser model.User, from, to *projectv1.Project,
+) error {
+	if !curUser.Admin && !from.Immutable && ((curUser.ID != model.UserID(from.UserId)) ||
+		(curUser.ID != model.UserID(to.UserId))) {
```

Comment:
Allowing someone who only owns the destination project allows anyone to make a new destination project and "steal" someones experiment from any project

Though I think the current protection was nothing so this might be fine?

Reply:
I will rewrite this in a second since the logic is confusing in the negative; I do mean to require from and to projects to be from the current UserId